src/clarity-angular/select/select-input.ts

```diff
@@ -27,30 +28,64 @@ export class SelectInput implements OnDestroy {
             }
         });
 
-        this._subInput = selectService.input.subscribe((value: string) => {
-            if (value !== (<HTMLInputElement>this.el).value) {
+        this._subInput = selectService.inputChange.subscribe((value: string) => {
+            if (value !== (<HTMLInputElement>this.el).value && value !== undefined && value !== null) {
                 (<HTMLInputElement>this.el).value = value;
             }
         });
     }
-
+    /**
+     * Notifies about keyup events for up down and enter
+     *
+     * @param {KeyboardEvent} e
+     * @memberof SelectInput
+     */
     @HostListener("keyup", ["$event"])
     onKeyUp(e: KeyboardEvent) {
         if (e.code.indexOf("Arrow") !== -1 || e.code.indexOf("Enter") !== -1) {
             this.selectService.specialKey = e.code;
-        } else {
-            this.selectService.input = (<HTMLInputElement>this.el).value;
+            if (e.code === "ArrowUp") {
+                if (this.ifOpenService.open) {
+                    this.selectService.focusPrevious();
+                }
+            } else if (e.code === "ArrowDown") {
+                if (!this.ifOpenService.open) {
+                    this.ifOpenService.open = true;
+                }
+                this.selectService.focusNext();
+            } else if (e.code === "Enter") {
+                if (this.ifOpenService.open && this.selectService.highlighted != null) {
+                    this.selectService.selectCurrentFocused();
+                    this.ifOpenService.open = false;
+                }
+            }
         }
     }
-
+    /**
+     * This HostListener exists to surpress the normal input behaviour on up down and enter.
+     *
+     * @param {KeyboardEvent} e
+     * @memberof SelectInput
+     */
     @HostListener("keydown", ["$event"])
     onKeyDown(e: KeyboardEvent) {
         if (e.code.indexOf("ArrowUp") !== -1 || e.code.indexOf("ArrowDown") !== -1 || e.code.indexOf("Enter") !== -1) {
             e.preventDefault();
         }
     }
+    /**
+     * Pushes value changes
+     *
+     * @param {Event} changes
+     * @memberof SelectInput
+     */
+    @HostListener("input", ["$event"])
+    onChange(changes: Event): void {
```

Comment:
If all you're using is the value of the input, just `@HostListener("input", ["$event.target.value"])`.